src/runtime-rs/crates/shimv2/src/shim_run.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::os::unix::io::RawFd;
+
+use anyhow::{Context, Result};
+
+use crate::{
+    logger,
+    shim::{ShimExecutor, ENV_KATA_BIND_FD},
+    Error,
+};
+
+impl ShimExecutor {
+    pub async fn run(&mut self) -> Result<()> {
+        crate::panic_hook::set_panic_hook();
+        let sid = self.args.id.clone();
+        let bundle_path = self.get_bundle_path().context("get bundle")?;
+        let path = bundle_path.join("log");
+        let _logger_guard =
+            logger::set_logger(path.to_str().unwrap(), &sid, self.args.debug).context("set logger");
+
+        self.do_run()
+            .await
+            .map_err(|err| {
+                error!(sl!(), "failed run shim {:?}", err);
+                err
+            })
+            .context("long run")?;
+
+        Ok(())
+    }
+
+    async fn do_run(&mut self) -> Result<()> {
+        info!(sl!(), "start to run");
+        self.args.validate(false).context("validata")?;
+
+        let _server_fd = get_server_fd().context("get server fd")?;
+        // TODO: implement run
+
+        Ok(())
+    }
+}
+
+fn get_server_fd() -> Result<RawFd> {
+    let env_fd = std::env::var(ENV_KATA_BIND_FD).map_err(Error::EnvVar)?;
+    let fd = env_fd.parse::<i32>().map_err(|_| Error::ServerFd(env_fd))?;
```

Comment:
Why does this accept a signed value?

Reply:
Thanks for your review, change i32 to RawFd